google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ReadWriteTransaction.java

```diff
@@ -396,26 +396,29 @@ public void onSuccess(ResultSet result) {}
   }
 
   @Override
-  public ApiFuture<ResultSetStats> analyzeUpdateAsync(
+  public ApiFuture<ResultSet> analyzeUpdateAsync(
       ParsedStatement update, AnalyzeMode analyzeMode, UpdateOption... options) {
-    return internalExecuteUpdateAsync(update, analyzeMode, options);
+    return ApiFutures.transform(
+        internalExecuteUpdateAsync(update, analyzeMode, options),
+        Tuple::y,
+        MoreExecutors.directExecutor());
   }
 
   @Override
   public ApiFuture<Long> executeUpdateAsync(
       final ParsedStatement update, final UpdateOption... options) {
     return ApiFutures.transform(
         internalExecuteUpdateAsync(update, AnalyzeMode.NONE, options),
-        ResultSetStats::getRowCountExact,
+        Tuple::x,
         MoreExecutors.directExecutor());
   }
 
-  private ApiFuture<ResultSetStats> internalExecuteUpdateAsync(
+  private ApiFuture<Tuple<Long, ResultSet>> internalExecuteUpdateAsync(
```

Comment:
I understand that this is internal, but can we add a comment somewhere to make it clear that we are setting only one of `Tuple::x` and `Tuple::y` and not both (even though both are available).

Reply:
Done

Reply:
That's a very detailed comment, thank you for it!